File: Practica-3/Pila.java
package pila;

class Pila {
	private float/*long*/ arreglo[];
	private int top;
	private int n;
	
	public Pila(int n) {
		this.n = n;
		arreglo = new float/*long*/[n + 1];
		this.top = 0;
	}
	public void push(/*long*/float e) {
		if (isFull()) {
			System.out.println("Pila llenaa");
		}else {
			this.top = this.top + 1;
			arreglo[this.top] = e;
		}
	}
	public float /*long*/ pop() {
		float dato = 0;
		dato = arreglo[this.top];
		this.top = this.top - 1;
		return dato;
	}
	public float /*long*/ peek() {
		return arreglo[this.top];
	}
	public boolean isEmpty() {
		return this.top == 0;
	}
	public boolean isFull() {
		return this.top == n;
	}
	
	public class principal {
		public static void main(String[] args) {
			System.out.println("aaaaaaaaaaaa");
			Pila z = new Pila(9);
			z.push(9);
			z.push(4);
			z.push(2);
			
			//z.mostrar(z);
		}
	}
	public static void mostrar(Pila pila) {
		Pila aux = new Pila(20);
		while(! pila.isEmpty()) {
			float dato = pila.pop();
			System.out.println(dato);
			aux.push(dato);
		}
		while(! pila.isEmpty()) {
			float dato = aux.pop();
			pila.push(dato);
		}
	}

	
}

File: Practica-3/Pila.py
class Pila:
    def __init__(self,n):
        self.__n = n
        self.__arreglo = [0] * n
        self.__top = -1
    def push(self, e):
        if not self.isFull():
            self.__top +=1
            self.__arreglo[self.__top] = e
        else:
            print("Pila llena")
    def pop(self):
        if self.isEmpty():
            print("pila vacia!!")
        else:
            elemento = self.__arreglo[self.__top]
            self.__top -= 1
            return elemento
    def peek(self):
        if self.isEmpty():
            print("Pila vacia")
        else:
            return self.__arreglo[self.__top]
    def isEmpty(self):
        return self.__top == -1
    def isFull(self):
        return self.__top == self.__n -1

    
p = Pila(9)
p.push(6)
p.push(23)
p.push(9)
p.push(7)
p.push(0)
p.push(1)
p.pop()
p.pop()


if p.isEmpty():
    print("Pila vacia")
else:
    print("Pila: ")
    for i in range(p._Pila__top + 1):
        print(p._Pila__arreglo[i])  
    print()
